src/components/Petitions/index.js

```diff
@@ -1,30 +1,45 @@
 import React from 'react';
-import settings from 'settings';
+import styles from './petitions.scss';
 import Container from 'components/Container';
 import TeaserGrid from 'components/TeaserGrid';
 import Header from 'components/Header';
 import PageTitle from 'components/PageTitle';
+import PetitionsFilters from 'components/PetitionsFilters';
 import NoResults from 'components/NoResults';
 import Pagination from 'containers/Pagination';
 
-const Petitions = ({ petitions, isLoading }) => (
+const Petitions = ({
+  params,
+  petitions,
+  isLoading,
+  title,
+  totalCount,
+  autocompleteProps
+}) => (
   <Container>
     <section>
-      <Header>
-        <PageTitle
-          title={settings.petitionsPage.title}
-          centered
-        />
-      </Header>
+      <div className={styles['header-wrapper']}>
+        <Header>
+          <PageTitle
+            title={title}
+            centered />
+
+          <PetitionsFilters
+            autocompleteProps={autocompleteProps}
+          />
+        </Header>
+      </div>
+
       {petitions.length || isLoading
         ? <TeaserGrid
           petitions={petitions}
           isLoading={isLoading}
           />
         : <NoResults />
       }
-      {petitions.length > 0 &&
-        <Pagination />
+
+      {petitions.length > 0 && totalCount > petitions.length &&
```

Comment:
seems a bit messy here? as well as @eschaefer added a check inside `Pagination` to check for rendering.

if it's absolutely crucial, add a `showPagination` boolean calculated in the container?